File: src/main/java/com/fish/entity/vo/AccountVO.java
package com.fish.entity.vo;

import com.fish.entity.pojo.Account;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@AllArgsConstructor
@NoArgsConstructor
public class AccountVO implements Serializable {
    private String userId;
    /**
     * 昵称
     */
    private String nickname;
    /**
     * 头像
     */
    private String avatar;
    /**
     * 性别
     */
    private String gender;
    /**
     * 专业
     */
    private String specialty;
    /**
     * 博客地址
     */
    private String blogAddress;
    /**
     * GitHub地址
     */
    private String githubAddress;
    public AccountVO(Account account) {
        this.userId = account.getUserId();
        this.nickname = account.getNickname();
        this.avatar = account.getAvatar();
        this.gender = account.getGender();
        this.specialty = account.getSpecialty();
        this.blogAddress = account.getBlogAddress();
        this.githubAddress = account.getGithubAddress();
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getNickname() {
        return nickname;
    }

    public void setNickname(String nickname) {
        this.nickname = nickname;
    }

    public String getAvatar() {
        return avatar;
    }

    public void setAvatar(String avatar) {
        this.avatar = avatar;
    }

    public String getGender() {
        return gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }

    public String getSpecialty() {
        return specialty;
    }

    public void setSpecialty(String specialty) {
        this.specialty = specialty;
    }

    public String getBlogAddress() {
        return blogAddress;
    }

    public void setBlogAddress(String blogAddress) {
        this.blogAddress = blogAddress;
    }

    public String getGithubAddress() {
        return githubAddress;
    }

    public void setGithubAddress(String githubAddress) {
        this.githubAddress = githubAddress;
    }
}

File: src/main/kotlin/com/fish/entity/vo/AccountVO.kt
package com.fish.entity.vo

import com.fish.entity.pojo.Account
import java.io.Serializable

data class AccountVO(
    val userId: String?,
    /**
     * 昵称
     */
    val nickname: String?,
    /**
     * 头像
     */
    val avatar: String?,
    /**
     * 性别
     */
    val gender: String?,
    /**
     * 专业
     */
    val specialty: String?,
    /**
     * 邮箱
     */
    val email: String?,
    /**
     * 博客地址
     */
    val blogAddress: String?,
    /**
     * GitHub地址
     */
    val githubAddress: String?
) : Serializable {
    /**
     * 角色
     */
    var role: String? = null
    constructor(account: Account) : this(
        account.userId,
        account.nickname,
        account.avatar,
        account.gender,
        account.specialty,
        account.email,
        account.blogAddress,
        account.githubAddress
    ) {
        if (account.roleId == 3L)
            this.role = "普通用户"
        else
            this.role = "管理员"
    }
}
